akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -243,17 +271,21 @@ import akka.annotation.InternalApi
 private[typed] class GuardianActorAdapter[T](_initialBehavior: Behavior[T]) extends ActorAdapter[T](_initialBehavior) {
   import Behavior._
 
+  def receive: Receive = waitingForStart(Nil)
+
   override def preStart(): Unit =
     if (!isAlive(behavior))
       context.stop(self)
-    else
-      context.become(waitingForStart(Nil))
 
   def waitingForStart(stashed: List[Any]): Receive = {
     case GuardianActorAdapter.Start =>
       start()
-
-      stashed.reverse.foreach(receive)
+      if (Behavior.isAlive(behavior)) {
+        context.become {
+          case msg => receiveMessage(msg)
+        }
+        stashed.reverse.foreach(receive)
```

Comment:
what is this doing? `receive` is `waitingForStart(Nil)`?

Reply:
Ups, yes, meant to unstash into the newly become behavior, not `receive`

Reply:
would it be less obscure to do?
```
stashed.reverse.foreach(receiveMessage)
```